internal/manager/manager_tasks.go

```diff
@@ -321,28 +323,38 @@ func (s *Manager) StashBoxBatchPerformerTag(ctx context.Context, input StashBoxB
 		}
 		box := boxes[input.Endpoint]
 
-		var tasks []StashBoxPerformerTagTask
+		var tasks []StashBoxBatchTagTask
 
 		// The gocritic linter wants to turn this ifElseChain into a switch.
 		// however, such a switch would contain quite large blocks for each section
 		// and would arguably be hard to read.
 		//
 		// This is why we mark this section nolint. In principle, we should look to
 		// rewrite the section at some point, to avoid the linter warning.
-		if len(input.PerformerIds) > 0 { //nolint:gocritic
+		if len(input.Ids) > 0 { //nolint:gocritic
+			// The user has chosen only to tag the items on the current page
 			if err := s.Repository.WithTxn(ctx, func(ctx context.Context) error {
 				performerQuery := s.Repository.Performer
 
-				for _, performerID := range input.PerformerIds {
+				for _, performerID := range input.Ids {
 					if id, err := strconv.Atoi(performerID); err == nil {
 						performer, err := performerQuery.Find(ctx, id)
 						if err == nil {
-							tasks = append(tasks, StashBoxPerformerTagTask{
-								performer:       performer,
-								refresh:         input.Refresh,
-								box:             box,
-								excluded_fields: input.ExcludeFields,
-							})
+							if err := performer.LoadStashIDs(ctx, performerQuery); err != nil {
+								return fmt.Errorf("loading performer stash ids: %w", err)
+							}
+
+							// Check if the user wants to refresh existing or new items
+							if (input.Refresh && len(performer.StashIDs.List()) > 0) ||
+								(!input.Refresh && len(performer.StashIDs.List()) == 0) {
+								tasks = append(tasks, StashBoxBatchTagTask{
+									performer:       performer,
+									refresh:         input.Refresh,
+									box:             box,
+									excluded_fields: input.ExcludeFields,
+									task_type:       Performer,
+								})
+							}
```

Comment:
Small refactor to load the StashIDs here instead of in the individual tasks. This way instead of adding a task for every single item and then deciding in the task if the user wanted to refresh existing or search for untagged, we are filtering here and only creating tasks for those items.